src/tools/jsondocck/src/config.rs

```diff
@@ -0,0 +1,36 @@
+use getopts::Options;
+
+#[derive(Debug)]
+pub struct Config {
+    /// The directory documentation output was generated in
+    pub doc_dir: String,
+    /// The file documentation was generated for, with docck commands to check
+    pub template: String,
+}
+
+pub fn parse_config(args: Vec<String>) -> Config {
+    let mut opts = Options::new();
+    opts.reqopt("", "doc-dir", "Path to the documentation directory", "PATH")
+        .reqopt("", "template", "Path to the template file", "PATH")
+        .optflag("h", "help", "show this message");
+
+    let (argv0, args_) = args.split_first().unwrap();
+    if args.len() == 1 || args[1] == "-h" || args[1] == "--help" {
+        let message = format!("Usage: {} <doc-dir> <template>", argv0);
+        println!("{}", opts.usage(&message));
+        std::process::exit(1);
+    }
+
+    let matches = opts.parse(args_).unwrap();
+
+    if matches.opt_present("h") || matches.opt_present("help") {
+        let message = format!("Usage: {} <doc-dir> <template>", argv0);
+        println!("{}", opts.usage(&message));
+        std::process::exit(1);
+    }
```

Comment:
Why do you check for `-h` twice?

Reply:
I just copied this code from compiletest. I think it's so that invalid args are ignored, if the first one is `-h`. I can change it if you want.

Reply:
I would prefer to only check it in one place, yeah.